x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.test.ts

```diff
@@ -0,0 +1,160 @@
+  beforeEach(() => {
+    responseMock.ok.mockClear();
+    responseMock.notFound.mockClear();
+    responseMock.customError.mockClear();
+    fetchMock.mockReset();
+  });
+  it('makes an API call and returns the response', async () => {
+    const appSearchAPIResponseBody = {
+      results: [{ name: 'engine1' }],
+      meta: { page: { total_results: 1 } },
+    };
+
+    AppSearchAPI.mockReturn(appSearchAPIResponseBody);
+
+    const requestHandler = createEnterpriseSearchRequestHandler({
+      config: mockConfig,
+      log: mockLogger,
+      path: '/as/credentials/collection',
+    });
+
+    await makeAPICall(requestHandler, {
+      query: {
+        type: 'indexed',
+        pageIndex: 1,
+      },
+    });
+
+    AppSearchAPI.shouldHaveBeenCalledWith(
+      'http://localhost:3002/as/credentials/collection?type=indexed&pageIndex=1',
+      {
+        headers: { Authorization: KibanaAuthHeader },
+      }
+    );
+
+    expect(responseMock.ok).toHaveBeenCalledWith({
+      body: appSearchAPIResponseBody,
+    });
+  });
+
+  describe('when an API request fails', () => {
+    it('should return 404 with a message', async () => {
+      AppSearchAPI.mockReturnError();
+
+      const requestHandler = createEnterpriseSearchRequestHandler({
+        config: mockConfig,
+        log: mockLogger,
+        path: '/as/credentials/collection',
+      });
+
+      await makeAPICall(requestHandler);
+
+      AppSearchAPI.shouldHaveBeenCalledWith(`http://localhost:3002/as/credentials/collection?`, {
+        headers: { Authorization: KibanaAuthHeader },
+      });
+
+      expect(responseMock.customError).toHaveBeenCalledWith({
+        body: 'Error connecting or fetching data from Enterprise Search',
+        statusCode: 502,
+      });
+    });
+  });
+
+  describe('when `hasValidData` fails', () => {
+    it('should return 404 with a message', async () => {
+      const appSearchAPIResponseBody = {
+        foo: 'bar',
+      };
+
+      AppSearchAPI.mockReturn(appSearchAPIResponseBody);
+
+      const requestHandler = createEnterpriseSearchRequestHandler({
+        config: mockConfig,
+        log: mockLogger,
+        path: '/as/credentials/collection',
+        hasValidData: (body?: IMockResponse) => {
+          return (
+            Array.isArray(body?.results) && typeof body?.meta?.page?.total_results === 'number'
+          );
+        },
+      });
+
+      await makeAPICall(requestHandler);
+
+      AppSearchAPI.shouldHaveBeenCalledWith(`http://localhost:3002/as/credentials/collection?`, {
+        headers: { Authorization: KibanaAuthHeader },
+      });
+
+      expect(responseMock.customError).toHaveBeenCalledWith({
+        body: 'Error connecting or fetching data from Enterprise Search',
+        statusCode: 502,
+      });
+    });
+  });
+});
+
+const makeAPICall = (handler: Function, params = {}) => {
+  return handler(
+    null,
+    {
+      headers: {
+        authorization: KibanaAuthHeader,
```

Comment:
Since we're setting this as a default in `MakeAPICall()`, IMO we can expect it as a default as well in `shouldHaveBeenCalledWith()`, instead of having to constantly pass that in manually per-test.

Example code diff:

https://github.com/JasonStoltz/kibana/commit/73343e247a1b978f15057cf39491829bc2c71ba3

Feel free to cherry-pick that commit, or if you're cool with it I can open a PR to your PR :)

Reply:
Good call, I like it

Reply:
Totally botched the cherry-pick so https://github.com/elastic/kibana/pull/75487/commits/c5cd224111e76059c1cf79bb1577b0388ce8d947 and https://github.com/elastic/kibana/pull/75487/commits/105038a703b8a34f4dc500c1de1ebfc89daacdf9